core/src/smartcontracts/isi/domain.rs

```diff
@@ -171,28 +173,26 @@ pub mod isi {
             let asset_definition_id = self.object;
 
             let mut assets_to_remove = Vec::new();
-            for (_, account) in state_transaction.world.accounts.iter() {
-                assets_to_remove.extend(
-                    account
-                        .assets
-                        .values()
-                        .filter_map(|asset| {
-                            if asset.id().definition == asset_definition_id {
-                                return Some(asset.id());
-                            }
-
-                            None
-                        })
-                        .cloned(),
-                )
-            }
+            assets_to_remove.extend(
+                state_transaction
+                    .world
+                    .assets
+                    .iter()
+                    .filter_map(|(asset_id, _asset)| {
+                        if asset_id.definition == asset_definition_id {
+                            return Some(asset_id);
+                        }
+                        None
+                    })
```

Comment:
```suggestion
                    .filter(|(asset_id, _)| asset_id.definition == asset_definition_id)
                    .map(|(asset_id, _)| asset_id)
```

Reply:
I think clippy will complain about `filter` followed by `map`

Reply:
https://github.com/rust-lang/rust-clippy/issues/9098

Reply:
Checking it, but I think it only does that if `filter` checks `is_some` or `is_ok` and `map` unwraps